frontend/src/metabase/actions/components/ActionViz/ActionVizForm.tsx

```diff
@@ -83,6 +85,28 @@ function ActionVizForm({
     setShowEditModal(false);
   };
 
+  const fetchInitialValues = useCallback(async () => {
+    const prefetchDashcardValues =
+      getDashboardType(dashboard.id) === "public"
+        ? PublicApi.prefetchDashcardValues
+        : ActionsApi.prefetchDashcardValues;
+
+    const canPrefetch = Object.keys(initialValues).length > 0;
+
+    if (!canPrefetch) {
+      return {};
+    }
+
+    return prefetchDashcardValues({
+      dashboardId: dashboard.id,
+      dashcardId: dashcard.id,
+      parameters: JSON.stringify(initialValues),
+    });
+  }, [dashboard.id, dashcard.id, initialValues]);
+
+  const shouldPrefetch =
+    action.type === "implicit" && action.kind === "row/update";
```

Comment:
there are two other places with the same check. probably it's a sign for having a helper

Reply:
86d158832b84fbca158158ea8fa1f6427e15d693 + 6c568b904cd6a2bba738ee3fbd925e59ce976d0c